bee-storage/bee-storage/src/storage.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use serde::de::DeserializeOwned;
+
+use std::error::Error;
+
+#[async_trait]
+/// Trait to be implemented on storage backend, which determine how to start and shutdown the storage.
+pub trait Backend: Sized + Send + Sync + 'static {
```

Comment:
I'ld prefer the term `StorageBackend` or just `Storage`. The reason is, that the term `Backend` is not specific enough as there are a lot of backends for all kinds of things out there. One could argue, that one only needs to import the module `storage`, and in the code one would always use `storage::Backend` which then is crystal clear when reading it. But if we tend to import the trait itself (which I think is more likely), then IMHO `StorageBackend` or `Storage` would be a tad nicer to read.